test/validator_test.rb

```diff
@@ -59,4 +59,46 @@ class Foo
       end
     end
   end
+
+  def test_validate_recursive_type_alias
```

Comment:
Please include assertions to let us know which `type` declaration is prohibited/allowed.

Reply:
All of the examples are prohibited ones. 👍